src/wp-includes/media.php

```diff
@@ -5564,6 +5686,123 @@ function wp_get_loading_attr_default( $context ) {
 	return 'lazy';
 }
 
+/**
+ * Gets loading optimization attributes.
+ *
+ * @since 6.3.0
+ *
+ * @param string  $tag_name The tag name.
+ * @param array[] $attr     Array of the attributes for the tag.
+ * @param string  $context  Context for the element for which the loading optimization attribute is requested.
+ *
+ * @return array[] Loading optimization attributes.
+ */
+function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
+	global $wp_query;
+
+	$loading_attrs = array();
+
+	// For now this function only supports images.
+	if ( 'img' !== $tag_name ) {
+		return $loading_attrs;
+	}
+
+	// For any resources, width and height must be provided, to avoid layout shifts.
+	if ( ! isset( $attr['width'] ) || ! isset( $attr['height'] ) ) {
+		return $loading_attrs;
+	}
+
+	if ( isset( $attr['loading'] ) && 'lazy' === $attr['loading'] ) {
+		if ( isset( $attr['fetchpriority'] ) && 'high' === $attr['fetchpriority']
+		) {
+			_doing_it_wrong(
+				__FUNCTION__,
+				sprintf(
+					/* translators: %s: fetchpriority="high". */
+					__( 'An image cannot be lazy-loaded and assigned %s at the same time.' ),
+					'<code>fetchpriority="high"</code>'
+				),
+				'6.3.0'
+			);
+		}
+
+		$loading_attrs['loading'] = 'lazy';
+		return wp_maybe_add_fetchpriority_high_attr( $loading_attrs, $attr );
```

Comment:
Follow up to the above comment, keep in mind this overall condition needs to be for _any_ custom-provided `loading` attribute (also see my original feedback in https://github.com/WordPress/wordpress-develop/pull/4495/files#r1207184790).
```suggestion
		if ( 'lazy' === $attr['loading'] ) {
			$loading_attrs['loading'] = $attr['loading'];
		}
		return wp_maybe_add_fetchpriority_high_attr( $loading_attrs, $attr );
```

Reply:
@felixarntz  I got this point. And also just to be on same page, 

we have this check `if ( 'lazy' === $attr['loading'] ) {` as the new function will only return loading='lazy' when applicable, it won't get updated by other values. Same is with fetchpriority='high'